src/clj/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -81,12 +81,18 @@
             options (if (nil? server)
                       {:servlet-init-params config}
                       {:servlet-init-params config :server-id (keyword server)})
-            auth-service (tk-services/maybe-get-service this :AuthorizationService)
-            auth-check-fn (if auth-service (partial tk-auth/authorization-check auth-service))]
-        (add-servlet-handler
-         (jolokia/create-servlet auth-check-fn)
-         route
-         options)))
+            require-auth? (get-in-config [:metrics :metrics-webservice :require-auth] true)
+            auth-service  (when require-auth? (tk-services/maybe-get-service this :AuthorizationService))
+            auth-check-fn (when auth-service (partial tk-auth/authorization-check auth-service))]
+        ;; When there should be authorization but there is no authorization service, fail fast
+        (when (and require-auth? (not auth-service))
+          (throw (ex-info (str "Config setting require-auth is set to true but authorization service was not found. "
+                               "If you don't require authorization, set the trapperkeeper-metrics config setting "
+                               "metrics.metrics-webservice.require-auth to false")
+                          {})))
```

Comment:
Instead of throwing here, should we use `get-service` instead of `maybe-get-service` above? Or are you wanting a more-specific error message than the one trapperkeeper would give you?